book/src/testnet-replicator.md

```diff
@@ -0,0 +1,200 @@
+##### Download Prebuilt Binaries
+If you would rather not use `solana-install` to manage the install, you can manually download and install the binaries.
+
+##### Linux
+Download the binaries by navigating to
+[https://github.com/solana-labs/solana/releases/latest](https://github.com/solana-labs/solana/releases/latest),
+download **solana-release-x86_64-unknown-linux-gnu.tar.bz2**, then extract the
+archive:
+```bash
+$ tar jxf solana-release-x86_64-unknown-linux-gnu.tar.bz2
+$ cd solana-release/
+$ export PATH=$PWD/bin:$PATH
+```
+##### mac OS
+Download the binaries by navigating to
+[https://github.com/solana-labs/solana/releases/latest](https://github.com/solana-labs/solana/releases/latest),
+download **solana-release-x86_64-apple-darwin.tar.bz2**, then extract the
+archive:
+```bash
+$ tar jxf solana-release-x86_64-apple-darwin.tar.bz2
+$ cd solana-release/
+$ export PATH=$PWD/bin:$PATH
+```
+##### Windows
+Download the binaries by navigating to
+[https://github.com/solana-labs/solana/releases/latest](https://github.com/solana-labs/solana/releases/latest),
+download **solana-release-x86_64-pc-windows-msvc.tar.bz2**, then extract it into a folder.
+It is a good idea to add this extracted folder to your windows PATH.
+
+##### Build From Source
+On Linux or mac OS, if you are unable to use the prebuilt binaries or prefer to build it yourself
+from source, navigate to
+[https://github.com/solana-labs/solana/releases/latest](https://github.com/solana-labs/solana/releases/latest),
+and download the **Source Code** archive.  Extract the code and build the
+binaries with:
+```bash
+$ ./scripts/cargo-install-all.sh .
+$ export PATH=$PWD/bin:$PATH
+```
+
+### Starting The Replicator
+Try running following command to join the gossip network and view all the other nodes in the cluster:
+```bash
+$ solana-gossip --entrypoint testnet.solana.com:8001 spy
+# Press ^C to exit
+```
+
+##### Windows
+Now configure the keypairs for your replicator by running:
+
+Navigate to the solana install location and open a cmd prompt
+```bash
+$ solana-keygen.exe new -o replicator-keypair.json
+$ solana-keygen.exe new -o storage-keypair.json
+```
+
+Use solana-keygen to show the public keys for each of the keypairs, 
+they will be needed in the next step:
+```bash
+# The replicator's identity
+$ solana-keygen.exe pubkey replicator-keypair.json
+$ solana-keygen.exe pubkey storage-keypair.json
+```
+
+Then set up the storage accounts for your replicator by running:
+```bash
+$ solana-wallet.exe --keypair replicator-keypair.json airdrop 100000
+$ solana-wallet.exe --keypair replicator-keypair.json create-replicator-storage-account <replicator_pubkey> <storage_account_pubkey>
+```
+
+To start the replicator:
+```bash
+$ solana-replicator.exe --entrypoint testnet.solana.com:8001 --identity replicator-keypair.json --storage-keypair storage-keypair.json --ledger replicator-ledger
+```
+
+If the mining pool public key is known then it can be passed in to the replicator to enable
```

Comment:
The mining pool setup is changing with @rob-solana’s work right?  Maybe we just don’t document reward collection until 0.17.

Reply:
Then they'll run out of tokens... :man_shrugging:

Reply:
How hard would it be to advertise the mining pool (genesis can already set it up)?

Reply:
Not hard, we could setup a well-known account. “StorageMiningPoo11111111111” or something.  But then we could also use that as the default pool

Reply:
(leveraging the recent primordial account work that Pankaj did)

Reply:
Nah I meant something for just v016 using the storage_contract::mining_pool. We could set it up in genesis and provide rpc or something. Then stick it in our instructions. Can get rid of it when we integrate with the proper rewards scheme in 0.17?

Reply:
Ah.  For 0.16 how about we add the file `multinode-demo/storage-mining-pool-keypair.json` into the repo, then use it by default in `multinode-demo/setup.sh`.   The pubkey for that keypair can then be the default value for the `--mining-pool` arg of solana-replicator.
Knowing the private key of a mining pool account doesn't give you anything once the pool is established in genesis.

Reply:
Yeah ill take a look at it today. Try to clean up this doc as well. We _might_ be able to do a fixed pub key like you suggested earlier since the private key is of no use to anyone anyway. I'll dig into it.

Reply:
Is the current impl ok? Got rid of the mining-pool argument and made it automatic like we discussed.